elyra/pipeline/validation.py

```diff
@@ -333,6 +333,14 @@ async def _validate_node_properties(self, pipeline_definition: PipelineDefinitio
                                                                        "nodeName": node_label,
                                                                        "propertyName": node_property,
                                                                        "keyName": key})
+                                link_ids = list(x.get('node_id_ref', None) for x in node.get_component_links)
+                                if node.get_component_parameter(node_property)['value'] not in link_ids:
+                                    response.add_message(severity=ValidationSeverity.Error,
```

Comment:
IIUC this only checks for direct parent nodes, but any upstream node is valid (ie parents of parents)

Reply:
ahh yes, thx for the reminder, leme change this to look recursively